kustomize/commands/edit/set/generatoroptions_test.go

```diff
@@ -0,0 +1,338 @@
+package set
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	testutils_test "sigs.k8s.io/kustomize/kustomize/v5/commands/internal/testutils"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func TestNewSetGeneratorOptionsIsNotNil(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	assert.NotNil(t, newCmdSetGeneratorOptions(fSys), nil)
+}
+
+func TestRunSetGeneratorOptionsValidate(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	cmd := newCmdSetGeneratorOptions(fSys)
+	testutils_test.WriteTestKustomization(fSys)
+	args := []string{}
+	cmd.SetArgs(args)
+
+	err := cmd.Execute()
+	assert.Error(t, err)
+	assert.Equal(t, "at least immutable, or disableNameSuffixHash or labels or annotations must be set.", err.Error())
+}
+
+func TestRunSetGeneratorOptionsWithImmutable(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	cmd := newCmdSetGeneratorOptions(fSys)
+	testutils_test.WriteTestKustomization(fSys)
+	args := []string{"--immutable"}
+	cmd.SetArgs(args)
+
+	err := cmd.Execute()
+	assert.NoError(t, err)
+
+	content, err := testutils_test.ReadTestKustomization(fSys)
+	assert.NoError(t, err)
+	expectedStr := strings.Join([]string{
+		"generatorOptions:",
+		"  immutable: true",
+	}, "\n")
+	assert.Contains(t, string(content), expectedStr)
+}
+
+func TestRunSetGeneratorOptionsWithDisableNameSuffixHash(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	cmd := newCmdSetGeneratorOptions(fSys)
+	testutils_test.WriteTestKustomization(fSys)
+	args := []string{"--disableNameSuffixHash"}
+	cmd.SetArgs(args)
+
+	err := cmd.Execute()
+	assert.NoError(t, err)
+
+	content, err := testutils_test.ReadTestKustomization(fSys)
+	assert.NoError(t, err)
+	expectedStr := strings.Join([]string{
+		"generatorOptions:",
+		"  disableNameSuffixHash: true",
+	}, "\n")
+	assert.Contains(t, string(content), expectedStr)
+}
+
+func TestRunSetGeneratorOptionsWithLabels(t *testing.T) {
+	type given struct {
+		args []string
+	}
+	type expected struct {
+		fileOutput []string
+		errMessage string
+	}
+	testCases := []struct {
+		description string
+		given       given
+		expected    expected
+	}{
+		{
+			description: "One pair",
+			given: given{
+				args: []string{
+					"--labels=a:b",
```

Comment:
Comments/help output uses `--labels=app=hazelcast` which one should be used?
`--labels=app:hazelcast`
```
  labels:
    app: hazelcast
```
vs `--labels=app=hazelcast`
```
  labels:
    app=hazelcast: ""
```

Reply:
`--labels=app:hazelcast` is correct. I fixed it :)